### PR TITLE
Setup React on Rails App

### PR DESCRIPTION
**Pull request by myaserkhan**
**In this PR implemented the following requirements:**

- Created a new Rails app.
- Created a table for storing messages and created 5 different greetings.
- Created an API endpoint that selects a random greeting from the table.
- Created a static view that will be the root of the app.
- Create the App and Greeting components.
- Created a store, an action, and a reducer that connects to the API endpoint to get the random greeting.
- Displayed the random greeting in the Greeting component.

### CODE REVIEW

File: app/javascript/components/App.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+function App() {
+  return (
+    <div className="App">
+      <h2>Someone says</h2>
+      <Greeting />
+    </div>
+  );
+}
+
+export default App;
```
Comment (GzimAsani) [inline app/javascript/components/App.js:13]: - [ ] `App`component needs to have a route that leads to the `Greeting` component, with routes (from react-router)
image

File: app/views/root/index.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<div id="root">
```
Comment (GzimAsani) [inline app/views/root/index.html.erb:1]: - [ ] There is an error coming from app/views/layouts/application.html.erb when loading the index page only, it is complaining that application.css does not exist, page works well when removing it, so please check this
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @myaserkhan                 :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your API, works great  :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/root/index.html.erb (hunk 2 above)
Comment (myaserkhan) [reply app/views/root/index.html.erb:1]: @GzimAsani the project is working fine with me.
Capture

Comment (GzimAsani) [inline app/views/root/index.html.erb:1]: - Hi @myaserkhan, ignore this request than, it must've been something wrong with my PC

**Review by rotimiazeez — APPROVED** [review]
Hi @myaserkhan👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good Functioning code 👍🏽
- No Linter error 👍🏽
- Good PR description 👍🏽
- Meaningful Commit messages 👍🏽

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**